scapy/contrib/isotp.py

```diff
@@ -168,6 +172,97 @@ def defragment(can_frames, use_extended_addressing=None):
         return results[0]
 
 
+class ISOTPHeader(CAN):
+    name = 'ISOTPHeader'
+    fields_desc = [
+        FlagsField('flags', 0, 3, ['error', 'remote_transmission_request', 'extended']),  # noqa: E501
+        XBitField('identifier', 0, 29),
+        ByteField('length', None),
+        ThreeBytesField('reserved', 0),
+    ]
+
+    def extract_padding(self, p):
+        return p, None
+
+    def post_build(self, p, pay):
+        if self.length is None:
+            p = p[:4] + chb(len(pay)) + p[5:]
+        return p + pay
+
+    def guess_payload_class(self, payload):
```

Comment:
Could you add a comment explaining the bit manipulation? For example, "The type is encoding in the first 4 bits of the first byte"